### PR TITLE
PhpIni Class

### LINKED ISSUE #651: PhpIni Class to manage all PHP-Ini stuff

### PR DESCRIPTION
**Pull request by ximex**
fixes #651

### CODE REVIEW

**Comment by ximex** [comment]
@Fasse please review and give feedback for improvements

File: adm_program/system/function.php [hunk 1]
```diff
@@ -841,3 +794,53 @@ function replaceValuesArrWithQM(array $valuesArray)
...
+    return PhpIni::getUploadMaxSize();
+}
+
+/**
+ * @deprecated 3.3.0:4.0.0 "admFuncGetBytesFromSize()" is deprecated, there is no replacement.
+ * @param string $data
+ * @param bool   $decimalMulti
+ * @return int
+ */
+function admFuncGetBytesFromSize($data, $decimalMulti = false)
```
Comment (Fasse) [inline adm_program/system/function.php]: Why not call the new method of the PhpIni class within this function? So we don't have the code twice.

File: adm_program/system/classes/phpini.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+        return ini_set('open_basedir', implode(PATH_SEPARATOR, $realDirPaths));
+    }
+
+    /**
+     * @param string $dirPath
+     * @return bool|string
+     * @throws Exception
+     */
+    public static function setFileUploadTmpDir($dirPath)
```
Comment (Fasse) [inline adm_program/system/classes/phpini.php]: This method is not called yet !?!
Please add an description of the method.

File: adm_program/system/classes/phpini.php [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+        return $isInBaseDirs;
+    }
+
+    /**
+     * @param string[] $dirPaths
+     * @return bool|string
+     * @throws Exception
+     */
+    public static function setBaseDirs(array $dirPaths = array())
```
Comment (Fasse) [inline adm_program/system/classes/phpini.php]: This method is not called yet !?!
Please add an description of the method.

File: adm_program/system/classes/phpini.php [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        return (self::getMemoryLimit() === -1 || self::getMemoryLimit() >= self::getPostMaxSize())
+            && (self::getPostMaxSize() === -1 || self::getPostMaxSize() >= self::getFileUploadMaxFileSize());
+    }
+
+    /**
+     * @param string $dirPath
+     * @return bool
+     */
+    private static function isInBaseDirs($dirPath)
```
Comment (Fasse) [inline adm_program/system/classes/phpini.php]: Please add an description of the method.

File: adm_program/system/classes/phpini.php [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+{
+    const BYTES_UNIT_FACTOR_1024 = 1024;
+    const BYTES_UNIT_FACTOR_1000 = 1000;
+
+    /**
+     * @param string $data
+     * @param int    $multi Factor to multiply. Default: 1024
+     * @return int
+     */
+    private static function getBytesFromSize($data, $multi = self::BYTES_UNIT_FACTOR_1024)
```
Comment (Fasse) [inline adm_program/system/classes/phpini.php]: Please add an description of the method. There should be a description of the expected format within $data

File: adm_program/system/classes/phpini.php (hunk 3 above)
Comment (ximex) [reply adm_program/system/classes/phpini.php]: I already have plans for this function in the future.

File: adm_program/system/classes/phpini.php (hunk 2 above)
Comment (ximex) [reply adm_program/system/classes/phpini.php]: I already have plans for this function in the future.

File: adm_program/system/function.php (hunk 1 above)
Comment (ximex) [reply adm_program/system/function.php]: because the function in PhpIni is private and so this doesn't work. and i want to leave the function private

**Pull request merged**